File: javascript_nodejs/21.custom-dialogs/dialogs/mainDialog/SlotDetails.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

class SlotDetails {

    /**
     * SlotDetails is a small class that defines a "slot" to be filled in a SlotFillingDialog.
     * @param {string} name The field name used to store user's response.
     * @param {string} promptId A unique identifier of a Dialog or Prompt registered on the DialogSet.
     * @param {string} prompt The text of the prompt presented to the user.
     * @param {string} reprompt (optional) The text to present if the user responds with an invalid value.
     */
    constructor(name, promptId, prompt, reprompt) {
        this.Name = name;
        this.PromptId = promptId;
        if (prompt && reprompt) {
            this.Options = {
                prompt: prompt,
                retryPrompt: reprompt,
            }
        } else {
            this.Options = prompt;
        }
    }
}

module.exports = SlotDetails;

File: samples/python/19.custom-dialogs/dialogs/slot_details.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from botbuilder.core import MessageFactory
from botbuilder.dialogs import PromptOptions


class SlotDetails:
    def __init__(
        self,
        name: str,
        dialog_id: str,
        options: PromptOptions = None,
        prompt: str = None,
        retry_prompt: str = None,
    ):
        self.name = name
        self.dialog_id = dialog_id
        self.options = (
            options
            if options
            else PromptOptions(
                prompt=MessageFactory.text(prompt),
                retry_prompt=None
                if retry_prompt is None
                else MessageFactory.text(retry_prompt),
            )
        )
